node/core/dispute-coordinator/src/participation/mod.rs

```diff
@@ -133,12 +138,13 @@ impl Participation {
 	/// The passed in sender will be used by background workers to communicate back their results.
 	/// The calling context should make sure to call `Participation::on_worker_message()` for the
 	/// received messages.
-	pub fn new(sender: WorkerMessageSender) -> Self {
+	pub fn new(sender: WorkerMessageSender, metrics: Metrics) -> Self {
 		Self {
 			running_participations: HashSet::new(),
-			queue: Queues::new(),
+			queue: Queues::new(metrics.clone()),
```

Comment:
Not related to your work but I wonder why it is okay to clone the prometheus instances here. I see it done in the examples and throughout our code but I don't understand why it works. What will happen if a metric is updated from two different instances?

Reply:
I had the same thought. So until I saw the other examples of this happening I tried passing through references to the same metrics handle. It made for very dirty code!

Reply:
If you look at the metrics, you'll see that they are `Arc`s, so it it is fine.

Reply:
Right... 
```
#[derive(Debug)]
pub struct GenericCounter<P: Atomic> {
    v: Arc<Value<P>>,
}
```

Thanks @eskimor